### ISSUE TITLE
some RST paragraphs are written out like lists

### ISSUE THREAD

**Issue by danse** [jgm/pandoc#4422]
paragraphs starting with the same prefix of a list will have the semantics of a list when written by pandoc, and can be represented as paragraphs by escaping the starting character. [from the doc:](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#enumerated-lists)
```
If a single-line paragraph begins with text identical to an enumerator ("A.", "1.", "(b)", "I)", etc.), the first character will have to be escaped in order to have the line parsed as an ordinary paragraph:

\A. Einstein was a really smart dude.
````

**Comment by danse** [comment]
an unit test is referenced above

**Comment by danse** [comment]
a simple solution here would be to import the list parsing logic into the writer and use it on paragraphs to detect whether escaping is needed.

an alternative would be to detect the condition *paragraph starting as a list* in the parser. there one could actually check for a *paragraph starting like anything else*, that is parse paragraph contents.

if we choose the latter we could change the parser to produce a `Div ("", [], [("escape", "first char")]) [Para [...`, which could be handled by the RST writer and any other writer, and could be extended in case of other ambiguities in the representation of paragraphs

**Comment by danse** [comment]
i imagine that it's a good idea to define keys and values for attributes somewhere so that they can be checked at compile-time, but i don't find where such definitions are contained in pandoc or pandoc-types. i guess that i could put them into `Pandoc.Shared`

**Comment by jgm** [comment]
You might check how the Markdown writer handles this
problem.

% echo '[Str "1.",Space,Str "hi"]' | pandoc -t rst -f native
1. hi
% echo '[Str "1.",Space,Str "hi"]' | pandoc -t markdown -f native
1\. hi


+++ Francesco Occhipinti [Mar 01 18 15:37 ]:

**Comment by danse** [comment]
good catch, the same problem has been tackled in the markdown writer. there the solution has been to [duplicate the parsing logic](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Writers/Markdown.hs#L355-L362) in the writer. i reckon the idea i had to detect such cases in the reader was wrong because the conditions of occurrence depend on the writing format. i will remove related comments from this issue.

still i wouldn't duplicate parsing logic, i'll try to import and reuse the logic from the reader

**Comment by jgm** [comment]
I prefer to limit what is exported from the readers, so it
might be better to do something a bit ad hoc in the writer.
